Mixpanel/SelectorEvaluator.swift

```diff
@@ -0,0 +1,604 @@
+    static let PLUS_OPERATOR = "+"
+    static let MINUS_OPERATOR = "-"
+    static let MUL_OPERATOR = "*"
+    static let DIV_OPERATOR = "/"
+    static let MOD_OPERATOR = "%"
+    static let EQUALS_OPERATOR = "=="
+    static let NOT_EQUALS_OPERATOR = "!="
+    static let GREATER_THAN_OPERATOR = ">"
+    static let GREATER_THAN_EQUAL_OPERATOR = ">="
+    static let LESS_THAN_OPERATOR = "<"
+    static let LESS_THAN_EQUAL_OPERATOR = "<="
+    // Unary operators
+    static let NOT_OPERATOR = "not"
+    static let DEFINED_OPERATOR = "defined"
+    static let NOT_DEFINED_OPERATOR = "not defined"
+    // Special words
+    static let NOW_LITERAL = "now"
+    
+    let selector: [String:Any]?
+    
+    init?(JSONObject: [String: Any]?) {
+        self.selector = JSONObject
+    }
+    
+    // For testing purposes
+    class func getCurrentDate() -> Date {
+        return Date();
+    }
+    
+    class func toNumber(value: Any?) -> Double? {
+        if let val = value {
+            switch val {
+            case let bool as Bool:
+                return bool ? 1 : 0
+            case let date as Date:
+                return date.timeIntervalSince1970 > 0 ? date.timeIntervalSince1970 : nil
+            case let num as NSNumber:
+                return num.doubleValue
+            case let str as String:
+                if let int = Int(str) {
+                    return Double(int)
+                }
+                if let double = Double(str) {
+                    return double
+                }
+                return 0.0
+            default:
+                return nil
+            }
+        }
+        return nil
+    }
+    
+    class func toBoolean(value: Any?) -> Bool {
+        if let val = value {
+            switch val {
+            case let bool as Bool:
+                return bool
+            case let num as NSNumber:
+                return num.doubleValue != 0 ? true : false
+            case let str as String:
+                return str.count > 0 ? true : false
+            case let arr as [Any?]:
+                return arr.count > 0 ? true : false
+            case let dict as [String: Any]:
+                return dict.count > 0 ? true : false
+            case let date as Date:
+                return date.timeIntervalSince1970 > 0 ? true : false
+            default:
+                return false
+            }
+        }
+        return false
+    }
+    
+    class func evaluateNumber(node: [String: Any], properties: Properties?) -> Double? {
+        guard let op = node[OPERATOR_KEY] as? String, op == NUMBER_OPERATOR else {
+            Logger.error(message: "invalid operator: number")
+            return nil
+        }
+        guard let children = node[CHILDREN_KEY] as? [[String: Any]], children.count == 1 else {
+            Logger.error(message: "invalid operator: number")
+            return nil
+        }
+        
+        return toNumber(value: evaluateNode(node: children[0], properties: properties))
```

Comment:
would be better to define const for `0` and `1` and avoid using magic number in this case